main.go

```diff
@@ -789,6 +806,16 @@ func main() {
 			}
 			metricAskpassCount.WithLabelValues(metricKeySuccess).Inc()
 		}
+
+		if *flGithubAppPrivateKey != "" && *flGithubAppInstallationID != -1 && *flGithubAppApplicationID != -1 {
+			if git.appTokenExpiry.Before(time.Now()) {
```

Comment:
Shouldn't there be some buffer here, so we don't fail TOCTOU ?  E.g. require max(20% of the token's lifespan, 30s) or something? 

I don't want tokens to be valid here but fail to sync 1 second later, and some of the ops we do can take O(seconds).

Reply:
That makes sense